querybook/webapp/lib/query-result/error-suggestion.ts

```diff
@@ -0,0 +1,97 @@
+import { IQueryEngine } from 'const/queryEngine';
+import {
+    IQueryError,
+    IQueryExecution,
+    IStatementExecution,
+} from 'const/queryExecution';
+
+const queryErrorsByLanguage: Record<
+    string,
+    Record<
+        string,
+        {
+            regex: string;
+            message: string;
+        }
+    >
+> = require('config/query_error.yaml');
+const SHARED_ERROR_SUGGESTION = 'common';
+
+// Merge all the common in
+for (const language of Object.keys(queryErrorsByLanguage)) {
+    if (language !== SHARED_ERROR_SUGGESTION) {
+        queryErrorsByLanguage[language] = {
+            ...queryErrorsByLanguage[language],
+            ...queryErrorsByLanguage[SHARED_ERROR_SUGGESTION],
+        };
+    }
+}
+
+const errorSuggestionInfoByLanguage = Object.entries(
+    queryErrorsByLanguage
+).reduce(
+    (hash, [language, queryErrors]) => {
+        if (language !== SHARED_ERROR_SUGGESTION) {
+            hash[language] = Object.entries(queryErrors).reduce(
+                (innerHash, [errorName, error]) => {
+                    innerHash[errorName] = {
+                        ...error,
+                        regex: new RegExp(error.regex, 'i'),
+                    };
+                    return innerHash;
+                },
+                {}
+            );
+        }
+        return hash;
+    },
+    {} as {
+        [language: string]: {
+            [errorName: string]: {
+                regex: RegExp;
+                message: string;
+            };
+        };
+    }
+);
+
+function getDefaultQueryErrorSuggestion(
+    queryError: IQueryError,
+    _queryExecution: IQueryExecution,
+    _statementExecutions: IStatementExecution[],
```

Comment:
seems these two parameters are not used, why need them?

Reply:
i did the nit part so I guess these parameters are required now 😂